File: python/sorting/sorting.py
class QuickSort:

    @staticmethod
    def sort(array):
        QuickSort.quicksort(array, 0, len(array) - 1)
        return array

    @staticmethod
    def quicksort(array, low, high):
        if low < high:
            pivot_index = QuickSort.partition(array, low, high)

            QuickSort.quicksort(array, low, pivot_index - 1)
            QuickSort.quicksort(array, pivot_index + 1, high)

    @staticmethod
    def partition(array, low, high):
        pivot = array[high]
        i = low - 1

        for j in range(low, high):
            if array[j] <= pivot:
                i += 1
                array[i], array[j] = array[j], array[i]

        array[i + 1], array[high] = array[high], array[i + 1]

        return i + 1
    
    

File: csharp/Sorting/Sorting.cs
namespace SortingNamespace
{
    public class QuickSort
    {
        public static int[] Sort(int[] array)
        {
            Quicksort(array, 0, array.Length - 1);

            return array;
        }

        private static void Quicksort(
            int[] array,
            int low,
            int high
        )
        {
            if (low < high)
            {
                int pivotIndex =
                    Partition(array, low, high);

                Quicksort(
                    array,
                    low,
                    pivotIndex - 1
                );

                Quicksort(
                    array,
                    pivotIndex + 1,
                    high
                );
            }
        }

        private static int Partition(
            int[] array,
            int low,
            int high
        )
        {
            int pivot = array[high];

            int i = low - 1;

            for (int j = low; j < high; j++)
            {
                if (array[j] <= pivot)
                {
                    i++;

                    (array[i], array[j]) =
                        (array[j], array[i]);
                }
            }

            (array[i + 1], array[high]) =
                (array[high], array[i + 1]);

            return i + 1;
        }
    }
}